initfiles/componentfiles/configschema/templates/AddVaults.json

```diff
@@ -0,0 +1,102 @@
+{
+    "name": "Add Vault Feature",
+    "description": "Add necessary keys with empty values for vault feature",
+    "type": "modification",
+    "operations": [
+        {
+            "action": "delete",
+            "target_path": "/Environment/Software/vaults",
+            "data": {
+                "error_if_not_found": false
+            }
+        },
+        {
+            "action": "create",
+            "target_path": "/Environment/Software",
+            "data": {
+                "node_type": "vaults",
+                "save": {
+                    "name" : "vaults_node"
+                }
+            }
+        },
+        {
+            "action": "create",
+            "description": "Create the git element in Environment/Software/vaults",
+            "target_nodeid": "{{vaults_node}}",
+            "data": {
+                "node_type": "git",
+                "attributes": [
+                    {
+                        "name": "name",
+                        "value": "git_vault"
+                    },
+                    {
+                        "name": "url",
+                        "value": "default"
+                    },
+                    {
+                        "name": "kind",
+                        "value": "kv_v2"
+                    },
+                    {
+                        "name": "client-secret",
+                        "value": "default"
```

Comment:
client-secret is actually optional.  I guess it needs to be filled in here so it shows up... but for this one it should tell them to either fill in a value or delete the tag.

Reply:
This has been modified to 

client-secret : optional_delete_if_unused